File: app/models/membership_history.py
from app.extensions import db

class MembershipHistory(db.Model):
    __tablename__ = 'membership_history'

    membership_history_id = db.Column(db.BigInteger, primary_key=True, autoincrement=True)
    action_type = db.Column(db.String(50), nullable=False)
    previous_status = db.Column(db.String(20), nullable=False)
    new_status = db.Column(db.String(20), nullable=False)
    history_note = db.Column(db.Text, nullable=True)

    #외래키
    membership_id = db.Column(db.BigInteger, db.ForeignKey('membership.membership_id'), nullable=False)

File: src/main/java/com/ticketflow/entity/MembershipHistory.java
package com.ticketflow.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "membership_history")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MembershipHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "membership_history_id")
    private Long membershipHistoryId;

    @Column(name = "action_type", length = 50, nullable = false)
    private String actionType;

    @Column(name = "previous_status", length = 20, nullable = false)
    private String previousStatus;

    @Column(name = "new_status", length = 20, nullable = false)
    private String newStatus;

    @Column(name = "history_note", columnDefinition = "TEXT")
    private String historyNote;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "membership_id", nullable = false)
    private Membership membership;
}
